hw-4/js/app.js

```diff
@@ -0,0 +1,176 @@
+
+function multiply(array) {
+    if (array.length !== 0) {
+        let result = 1;
+        for(let i = 0; i < array.length; i++) {
+            result *= array[i];
+        }
+        return result;
+    } else {
+        return 0;
+    }
+};
+
+// 2. Создать функцию, которая принимает строку и возвращает строку-перевертыш: reverseString(‘test’) // “tset”.
+
+console.log('---------- 2 ----------');
+var string2 = prompt("2. Please enter string:", "");
+var stringResult2 = reverseString(string2);
+
+console.log('Old string is ->', string2);
+console.log('New string is ->', stringResult2);
+
+function reverseString(string) {
+    let stringResult = '';
+    for (let i = string.length - 1; i >= 0; i--) {
+        stringResult += string[i];
+    }
+    return stringResult;
+}
+
+// 3. Создать функцию, которая в качестве аргумента принимает строку из букв и возвращает строку, где каждый символ разделен пробелом и заменен на юникод-значение символа:
+// getCodeStringFromText(‘hello’) // “104 101 108 108 111”
+
+console.log('---------- 3 ----------');
+var string3 = prompt("3. Please enter string:", "");
+var stringResult3 = getCodeStringFromText(string3);
+
+console.log('Old string is ->', string3);
+console.log('New string is ->', stringResult3);
+
+function getCodeStringFromText(string) {
+    let stringResult = '';
+    for(let i = 0; i < string.length; i++) {
+        stringResult += string.charCodeAt(i) + ' ';
+    }
+    return stringResult;
+}
+
+// 4. Создать функцию угадай число. Она принимает число от 1-10 (обязательно проверить что число не больше 10 и не меньше 0).
+// Генерирует рандомное число от 1-10 и сравнивает с переданным числом если они совпали то возвращает “Вы выиграли” если нет то “Вы не угадали
+// ваше число 8 а выпало число 5”. Числа в строке указаны как пример вы подставляете реальные числа.
+
+console.log('---------- 4 ----------');
+var number4 = prompt("4. Please enter number from 1 to 10:", "");
+guessNumber(number4);
+
+function guessNumber(number) {
+    let min = 1, max = 10;
+    let randomNumber = Math.round(Math.random() * (max-min)) + min;
+    if(number >= min && number <= max) {
+        if(randomNumber === +number) {
+            console.log('You win !!!');
+            console.log('Your number ->', number, 'Random number ->', randomNumber);
+        } else {
+            console.log('You lose !!!');
+            console.log('Your number ->', number, 'Random number ->', randomNumber);
+        }
+    } else {
+        console.log('You enter wrong number');
+    }
+}
+
+// 5. Создать функцию, которая принимает число n и возвращает массив, заполненный числами от 1 до n: getArray(10); // [1,2,3,4,5,6,7,8,9,10]
+
+console.log('---------- 5 ----------');
+var number5 = prompt("5. Please enter number:", "");
+var array5 = getArray(number5);
+
+console.log('Your number ->', number5);
+console.log('Array ->', array5);
+
+function getArray(number) {
+    let array = [];
+    for(let i = 0; i < +number; i++) {
```

Comment:
better check type in the beginning of function. Don't make it number in `for`. It looks confusing